File: android/src/main/java/com/befovy/fijkplayer/FijkRecorderHandler.java
package com.befovy.fijkplayer;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;

/**
 * Recorder Handler - delegates to FijkFFmpegRecorder for actual recording
 * 
 * Channel: befovy.com/fijk/recorder
 */
public class FijkRecorderHandler implements MethodChannel.MethodCallHandler {
    
    private static final String TAG = "FijkRecorderHandler";
    private final Context context;
    private FijkFFmpegRecorder recorder;
    
    public FijkRecorderHandler(Context context) {
        this.context = context;
        this.recorder = new FijkFFmpegRecorder();
        Log.i(TAG, "FijkRecorderHandler initialized");
    }
    
    @Override
    public void onMethodCall(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Log.d(TAG, "Method call: " + call.method);
        
        switch (call.method) {
            case "isRecording":
                result.success(recorder.isRecording());
                break;
            case "startRecording": {
                String rtspUrl = call.argument("rtspUrl");
                String outputPath = call.argument("outputPath");
                if (rtspUrl == null || outputPath == null) {
                    result.error("INVALID_ARGS", "Missing rtspUrl or outputPath", null);
                    return;
                }
                try {
                    boolean started = recorder.startRecording(rtspUrl, outputPath);
                    result.success(started);
                } catch (Exception e) {
                    Log.e(TAG, "Failed to start recording", e);
                    result.error("RECORDING_FAILED", e.getMessage(), null);
                }
                break;
            }
            case "stopRecording": {
                try {
                    boolean stopped = recorder.stopRecording();
                    result.success(stopped);
                } catch (Exception e) {
                    Log.e(TAG, "Failed to stop recording", e);
                    result.error("STOP_FAILED", e.getMessage(), null);
                }
                break;
            }
            default:
                Log.w(TAG, "Unimplemented recorder method: " + call.method);
                result.notImplemented();
                break;
        }
    }
}

File: ios/Classes/FijkRecorderHandler.m
//
//  FijkRecorderHandler.m
//  fijkplayer
//
//  Handler for befovy.com/fijk/recorder channel
//  Delegates to FFmpegRecorder for actual recording
//

#import "FijkRecorderHandler.h"
#import "FFmpegRecorder.h"

@implementation FijkRecorderHandler

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    NSLog(@"[FijkRecorderHandler] Method call: %@", call.method);
    
    if ([@"startRecording" isEqualToString:call.method]) {
        NSString *rtspUrl = call.arguments[@"rtspUrl"];
        NSString *outputPath = call.arguments[@"outputPath"];
        
        if (!rtspUrl || !outputPath) {
            result([FlutterError errorWithCode:@"INVALID_ARGS"
                                       message:@"Missing rtspUrl or outputPath"
                                       details:nil]);
            return;
        }
        
        NSError *error = nil;
        BOOL started = [[FFmpegRecorder sharedInstance] startRecordingWithRtspUrl:rtspUrl
                                                                      outputPath:outputPath
                                                                           error:&error];
        if (started) {
            result(@YES);
        } else {
            result([FlutterError errorWithCode:@"RECORDING_FAILED"
                                       message:error.localizedDescription ?: @"Failed to start recording"
                                       details:nil]);
        }
    } else if ([@"stopRecording" isEqualToString:call.method]) {
        NSError *error = nil;
        BOOL stopped = [[FFmpegRecorder sharedInstance] stopRecordingWithError:&error];
        if (stopped) {
            result(@YES);
        } else {
            result([FlutterError errorWithCode:@"STOP_FAILED"
                                       message:error.localizedDescription ?: @"Failed to stop recording"
                                       details:nil]);
        }
    } else if ([@"isRecording" isEqualToString:call.method]) {
        result(@([[FFmpegRecorder sharedInstance] isRecording]));
    } else {
        result(FlutterMethodNotImplemented);
    }
}

@end
